doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+
+MQTT doesn't mandate the data serialization format, so any application might implement its own format. Data serialization might be a tricky task and protocols might be hard to design, Astarte MQTT takes care of this and provides a higher level protocol which abstracts this detail from the end user.
+
+Astarte MQTT v1 Protocol builds upon MQTT v3.1.1 itself, [BSON](http://bsonspec.org/spec.html) (Binary JSON, version 1.1) serialized payloads and on optional zlib deflate. All communications are ordered and asynchronous.
+
+A protocol reference implementation is provided with an Astarte SDK, however developers might implement it from scratch using 3rd party libraries with their favourite languages: all formats and protocols described here are open and well documented. Last but not least Astarte doesn't mandate this protocol, and a different one can be used with a different transport.
+
+## MQTT Topics Overview
+
+Astarte MQTT v1 Protocol relies on few well known topics that are reserved.
+
+| Topic                                                  | Purpose          | Published By | QoS     | Payload Format                          |
+|--------------------------------------------------------|------------------|--------------|---------|-----------------------------------------|
+| `<realm name>/<device id>`                             | Introspection    | Device       | 2       | ASCII plain text, ':' and ';' delimited |
+| `<realm name>/<device id>/control/emptyCache`          | Empty Cache      | Device       | 2       | ASCII plain text (always "1")           |
+| `<realm name>/<device id>/control/consumer/properties` | Purge Properties | Astarte      | 2       | deflated plain text                     |
+| `<realm name>/<device id>/control/producer/properties` | Purge Properties | Device       | 2       | deflated plain text                     |
+| `<realm name>/<device id>/<interface name>/<path>`     | Publish Data     | Both         | 0, 1, 2 | BSON (or empty)                         |
+
+For clarity reasons all `<realm name>/<device id>` prefixes will be omitted on the following paragraphs, those topics will be called device topics.
+Topics are not bidirectional, devices must not publish data for server owned topics and viceversa, onwership is explicitly stated in interfaces files.
+
+## BSON
+
+BSON format allows to save precious bytes compared to JSON, while it offers the advantages of a schema-less protocol.
+Previous statement can be explained with two really simple examples, which are the same payload encoded with JSON and BSON:
```

Comment:
Rephrase and reduce:

Consider, for example, a simple value and timestamp payload. The encoded JSON version, `{"v":25.367812,"t":1537346756844}` counts 33 bytes.
The hex dump of the same message encoded with BSON is